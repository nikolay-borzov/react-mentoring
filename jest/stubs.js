export const film = {
  id: 1,
  title: 'Film title',
  poster_path: 'http://via.placeholder.com/500x750',
  overview: 'Film overview',
  release_date: '2018-05-04',
  genres: ['some genre', 'another genre']
}

export const films = [film]
