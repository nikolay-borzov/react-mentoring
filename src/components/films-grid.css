.film-grid {
  display: grid;
  grid-gap: 3.5em;
  grid-template-columns: repeat(auto-fit, minmax(250px, 300px));
  justify-content: center;
  padding: 2em 3em 3.5em;
}

@media (min-width: 1200px) {
  .film-grid {
    padding: 2em 10% 3.5em;
  }
}
