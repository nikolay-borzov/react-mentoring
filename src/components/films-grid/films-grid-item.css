@import '../../styles/variables.css';

.film-grid-item {
  display: flex;
  flex-direction: column;
  color: var(--color-default);
  text-decoration: none;

  @apply --hover-effect;
}

.film-grid-item__cover {
  max-width: 100%;
  display: block;
}

.film-grid-item__info {
  margin-top: 0.5em;
}

.film-grid-item__row {
  padding: 0.4em 0.5em;
}

.film-grid-item__title {
  display: flex;
  justify-content: space-between;
  align-items: start;
}

.film-grid-item__date {
  border: solid 1px var(--color-alt);
  padding: 0.1em 0.5em;
  border-radius: 0.2em;
  margin-left: 1em;
}
