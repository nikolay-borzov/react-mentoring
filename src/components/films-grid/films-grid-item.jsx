// @flow

import React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'

import { ContentImage } from '../'

type FilmsGridItemProps = {
  film: Film
}

const FilmGridItem = styled(Link)`
  display: flex;
  flex-direction: column;
  color: var(--color-default);
  text-decoration: none;
`

const FilmGridItemInfo = styled.div`
  margin-top: 0.5rem;
`

const FilmGridItemRow = styled.div`
  padding: 0.4rem 0.5rem;
`

const FilmGridItemTitle = styled(FilmGridItemRow)`
  display: flex;
  justify-content: space-between;
  align-items: start;
`

const FilmGridItemDate = styled.span`
  border: solid 1px var(--color-alt);
  padding: 0.1rem 0.5rem;
  border-radius: 0.2rem;
  margin-left: 1rem;
`

export function FilmsGridItem(props: FilmsGridItemProps) {
  const {
    film: {
      id,
      poster_path: posterPath,
      title,
      overview,
      release_date: releaseDate
    }
  } = props

  return (
    <FilmGridItem
      to={`/film/${id}`}
      className="hover-effect"
      tabIndex={0}
      data-cy="film-grid-item">
      <ContentImage src={posterPath} alt={title} title={overview} />

      <FilmGridItemInfo>
        <FilmGridItemTitle>
          <span className="uppercase font-bold">{title}</span>
          <FilmGridItemDate>{releaseDate.substring(0, 4)}</FilmGridItemDate>
        </FilmGridItemTitle>

        <FilmGridItemRow className="color-alt">
          {props.film.genres.join(', ')}
        </FilmGridItemRow>
      </FilmGridItemInfo>
    </FilmGridItem>
  )
}
