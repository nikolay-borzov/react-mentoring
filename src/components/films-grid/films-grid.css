@import '../../styles/variables.css';

.film-grid {
  display: grid;
  grid-gap: 3.5em;
  grid-template-columns: repeat(auto-fit, minmax(250px, 300px));
  justify-content: center;
  padding: var(--padding-content);
}
