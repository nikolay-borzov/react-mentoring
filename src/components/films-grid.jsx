import React from 'react'
import PropTypes from 'prop-types'

import './films-grid.css'

import { FilmsGridItem } from './films-grid-item'

export class FilmsGrid extends React.PureComponent {
  static propTypes = {
    films: PropTypes.arrayOf(PropTypes.object).isRequired
  }

  render() {
    return (
      <div className="film-grid">
        {this.props.films.map(film => (
          <FilmsGridItem film={film} key={film.id} />
        ))}
      </div>
    )
  }
}
