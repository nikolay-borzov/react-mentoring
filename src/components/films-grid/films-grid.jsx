// @flow

import React from 'react'

import './films-grid.css'

import { FilmsGridItem } from './films-grid-item'

type FilmsGridProps = {
  films: Array<Object>
}

export function FilmsGrid(props: FilmsGridProps) {
  return (
    <div className="film-grid padding-content" data-cy="film-grid">
      {props.films.map(film => <FilmsGridItem film={film} key={film.id} />)}
    </div>
  )
}

export default FilmsGrid
