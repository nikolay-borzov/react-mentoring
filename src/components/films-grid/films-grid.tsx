import React from 'react'
import styled from 'styled-components'

import { FilmsGridItem } from './films-grid-item'
import { Film } from '../../entities/film'

const FilmGrid = styled.div`
  display: grid;
  grid-gap: 3.5rem;
  grid-template-columns: repeat(auto-fit, minmax(250px, 300px));
  justify-content: center;
`

export interface FilmsGridProps {
  films: Film[]
}

export const FilmsGrid: React.FunctionComponent<FilmsGridProps> = props => (
  <FilmGrid className="padding-content" data-cy="film-grid">
    {props.films.map(film => (
      <FilmsGridItem film={film} key={film.id} />
    ))}
  </FilmGrid>
)

export default FilmsGrid
