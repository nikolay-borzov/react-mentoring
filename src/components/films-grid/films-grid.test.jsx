import React from 'react'

import {
  itRendersCorrectly,
  itContainsComponent
} from '../../../jest/test-helpers'
import { film, films } from '../../../jest/stubs'

import { FilmsGrid } from './films-grid'

describe('FilmsGrid component', () => {
  let props = {
    films: films
  }

  itRendersCorrectly(() => {
    return <FilmsGrid {...props} />
  })

  itContainsComponent(() => <FilmsGrid {...props} />, 'FilmsGridItem', {
    expectedProps: {
      film: film
    }
  })
})
