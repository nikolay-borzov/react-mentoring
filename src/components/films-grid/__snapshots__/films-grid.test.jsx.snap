// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsGrid component renders correctly 1`] = `
<div
  className="film-grid padding-content"
  data-cy="film-grid"
>
  <FilmsGridItem
    film={
      Object {
        "genres": Array [
          "some genre",
          "another genre",
        ],
        "id": 1,
        "overview": "Film overview",
        "poster_path": "http://via.placeholder.com/500x750",
        "release_date": "2018-05-04",
        "title": "Film title",
      }
    }
    key="1"
  />
</div>
`;
